lib/sinatra/base.rb

```diff
@@ -179,7 +175,7 @@ def drop_content_info?
     end
 
     def drop_body?
-      DROP_BODY_RESPONSES.include?(status.to_i)
+      !headers["Content-Type"] || DROP_BODY_RESPONSES.include?(status.to_i)
```

Comment:
It's not bad in general, but there are usecases for sinatra where it would really hurt. I suggest adding an option for this as well. I.e.:
```ruby
settings.drop_body_for_empty_content_type? && !headers["Content-Type"]
```